### ISSUE TITLE
Add support for internal version checks in HTEX

### ISSUE THREAD

**Issue by yadudoc** [Parsl/parsl#2957]
**Is your feature request related to a problem? Please describe.**
As Globus compute moves to using Parsl's HTEX managers often run different python and parsl versions than the interchange. Currently, this raises a `VersionMismatch` exception since HTEX requires a version match to avoid the several different errors that could creep in from version mismatch. In the GC context, this is less of a concern and an option to disable this check is preferrable.

**Describe the solution you'd like**
A kwarg option for the `HighThroughputExecutor(disable_internal_version_checks: bool = False)` which defaults to the current behavior.

**Comment by benclifford** [comment]
I think disabling that check, rather than putting some effort into protocol work, is an approach that will come back to haunt you :ghost: 

"disable/enable" features look nicer (to me) with fewer negatives, and phrased about whether the check should be made or not, rather than phrased as a change against soon-forgotten built-in behaviour: eg. instead of "disable_internal_version_checks: bool = False", have "check_version_mismatches: bool = True".

**Comment by yadudoc** [comment]
I was going to recommend [Messagepack](https://github.com/funcx-faas/funcx-common/tree/main/src/globus_compute_common/messagepack). I agree that doing this properly now is the right way to go.

**Comment by benclifford** [comment]
I don't think it brings in any additional requirements, but it's also interesting maybe to think about this happening inside Parsl: any of the places where we're interested in multiple environments existing - which is cloud stuff, not so interesting any more, but also WQ and radical which have the ability to build their own environments, and places where people are using containers.  I think that is broadly the same situation as with Globus Compute and doesn't change anything here, but I'm noting it on this issue anyway.
